### ISSUE TITLE
[BUG] Unable to extract TIFF files at 40x effective magnification

### ISSUE THREAD

**Issue by andrewsris** [jamesdolezal/slideflow#188]
## Description
I have slides that are scanned on a Philips scanner and saved as a TIFF file. I am able to extract at tile_px=299 / tile_um=301 successfully, but when I try to do higher effective magnification levels at 40x (i.e. tile_px=299 / tile_um=75), I receive the following repeating error message:

> (process:77602): GLib-GObject-WARNING **: 10:46:07.052: value "0" of type 'gint' is invalid or out of range for property 'width' of type 'gint'
> (process:77647): GLib-GObject-WARNING **: 10:46:07.052: value "0" of type 'gint' is invalid or out of range for property 'width' of type 'gint'
> (process:77602): GLib-GObject-WARNING **: 10:46:07.052: value "0" of type 'gint' is invalid or out of range for property 'height' of type 'gint'
> (process:77647): GLib-GObject-WARNING **: 10:46:07.052: value "0" of type 'gint' is invalid or out of range for property 'height' of type 'gint'

## To Reproduce
1.I will try to find a TIFF file publicly available to recreate the issue. 

`P.extract_tiles(tile_px=299, tile_um=75,
                    buffer='/ssd',
                    filters={"site": ["Sinai"]},
                    roi_method="auto",
                    qc='both', 
                    dry_run=False)`

## Environment:
Slideflow Version (e.g., 1.0): 1.2.4
OS (e.g., Ubuntu): Ubuntu
How you installed PyTorch (pip, source): pip3
Python version: 3.8
CUDA/cuDNN version:
GPU models and configuration: V100 x 4
Any other relevant information:

## Additional context
I will try to reach out to you directly to show you the issue.

**Comment by jamesdolezal** [comment]
This error comes from libvips attempting to open the slide image and encountering an error. My guess is that it's an issue with the image itself. I would start by checking if libvips can open the file on its own:

```python 
import pyvips

image = pyvips.Image.new_from_file(path)

print(image.height, image.width)
```

**Comment by andrewsris** [comment]
Thanks James. It looks like pyvips can can load the file.

> >>> image = pyvips.Image.new_from_file("MS-20-18311.tiff")
>>> print(image.height, image.width)
97792 135680
>>>

**Comment by andrewsris** [comment]
I will do more digging, but it looks like there are multiple people with similar problems in the libvips Issues Github. https://github.com/libvips/libvips/issues/2779

**Comment by jamesdolezal** [comment]
Ok - it would be easier to troubleshoot if you can find a publicly available slide image that reproduces the problem. Barring that, the next thing I would check would be to see if the image properties are loaded successfully, and then ensure each downsample layer in the pyramidal TIFF file is loaded appropriately. The `sf.slide._VIPSWrapper` class handles most of this automatically:

```python
from slideflow.slide import _VIPSWrapper

slide = _VIPSWrapper(path)
print(slide.levels)
```

My guess is that is going to raise the error. If so, we can dive a bit deeper into the libvips API to get more information:

```python
import pyvips
from slideflow.slide import (OPS_LEVEL_COUNT, 
                             OPS_WIDTH, 
                             OPS_HEIGHT,
                             OPS_LEVEL_HEIGHT,
                             OPS_LEVEL_WIDTH,
                             OPS_LEVEL_DOWNSAMPLE)

# Load the image and properties
print("Loading image and properties...")
image = pyvips.Image.new_from_file(path)
properties = {}
for field in image.get_fields():
    properties.update({field: image.get(field)})
dimensions = (
    int(properties[OPS_WIDTH]),
    int(properties[OPS_HEIGHT])
)

# Load each downsample level manually
print("Loading downsample levels...")
if OPS_LEVEL_COUNT in properties:
    level_count = int(properties[OPS_LEVEL_COUNT])
    levels = []
    for lev in range(level_count):
        width = int(image.get(OPS_LEVEL_WIDTH(lev)))
        height = int(image.get(OPS_LEVEL_HEIGHT(lev)))
        downsample = float(image.get(OPS_LEVEL_DOWNSAMPLE(lev)))
        levels += [{
            'dimensions': (width, height),
            'width': width,
            'height': height,
            'downsample': downsample
        }]
else:
    level_count = 1
    levels = [{
            'dimensions': dimensions,
            'width': dimensions[0],
            'height': dimensions[1],
            'downsample': 1
        }]

# Print the information we obtained, if no error raised
print("Image dimensions:   ", dimensions)
print("Number of levels:   ", level_count)
print("Level information:  ", levels) 
```

**Comment by andrewsris** [comment]
Hi James, 

I am using your most recent slideflow docker image, which is vips-8.12.2-Tue Jan 25 09:34:32 UTC 2022.

I have not done your most recent suggestion yet, but I was able to find a publicly available slide that recreates the issue (https://github.com/openslide/openslide/issues/145#issuecomment-510107828). I have sent you an email with a zipped entire slideflow project folder with this example slide to show you the bug.

In addition, here is Openslide's documentation about Philip's exported TIFF file, in case it is helpful. (https://openslide.org/formats/philips/)

I will keep you updated.

**Comment by jamesdolezal** [comment]
The issue was caused by the grayspace/whitespace filtering step using the lowest-level magnification image in the pyramid stack for filtering. With Philips scanner images, the lowest downsample level may contain too small of an image to use for grayspace filtering. I have revised the filtering step to only use valid pyramid layers containing a sufficiently large image.

This fix is in 3d1c126 and will be included in the 1.2.6 patch and version 1.3 onward.
